core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/PhysicalPlan.scala

```diff
@@ -63,6 +66,8 @@ case class PhysicalPlan(
     jgraphtDag
   }
 
+  @transient lazy val maxChains: Set[Set[PhysicalLink]] = this.getMaxChains
```

Comment:
To avoid having to compute this multiple times during the search I am saving this variable. The previous issue was I was not using this variable but calling the method. Fixed this issue.